src/ddmd/func.d

```diff
@@ -5294,9 +5294,18 @@ extern (C++) final class InvariantDeclaration : FuncDeclaration
  */
 extern (C++) static Identifier unitTestId(Loc loc)
 {
+    // there's a counter per module in case there are two unit tests on one line
+    static __gshared int[const(char)*] counterPerModule;
```

Comment:
@klickverbot 

> You are quite far from the point where "I don't want AA's [sic] in DMD" is enough of a justification – it wouldn't even be for Walter. ;)

I am not sure if this was meant jokingly or condescending ... that aside.

relying on d-runtime is not great because then the performance of the compiler is determined be the host druntime...

though I guess for a feature such as this it should not have too much of a impact.